blood/src/main/java/cz/cuni/mff/d3s/blood/dependencyMatrix/DependencyMatrixCollector.java

```diff
@@ -70,4 +122,53 @@ public static void postPhase(StructuredGraph graph, Class<?> sourceClass) {
         // currently, that's unnecessary, because we are using Graal's
         // own creation tracking
     }
+
+    private static void dump() {
+        // FIXME race condition
+        // graal apparently doesn't stop compiling before this shutdown hook gets called
+        
+        Date started = new Date();
+
+        final HashSet<Class> keys = new HashSet<>(HASHMAP_INIT_CAPACITY);
+        dependencyTable.forEachEntry(1, entry -> {
+            keys.add(entry.getKey());
+            entry.getValue().forEachKey(1, keys::add);
+        });
+        final Class[] keysOrder = keys.toArray(new Class[keys.size()]);
+
+        try (final OutputStreamWriter out = new OutputStreamWriter(new FileOutputStream("/tmp/gcopdd-depmat"))) {
+            // List of classes in the order that is used in the matrix below.
+            // Each line contains space-separated list of the class's
+            // superclasses from itself up to java.lang.Object.
+            Arrays.stream(keysOrder)
+                    .map(clazz
+                            -> Stream.iterate(clazz, NON_NULL, Class::getSuperclass)
+                            .map(Class::getName)
+                            .collect(Collectors.joining(" ", "", "\n"))
+                    )
+                    .forEachOrdered((CheckedConsumer<String>) out::append);
+
+            // Empty line.
+            out.append('\n');
+
+            // Dependency matrix. Lines correspond to rows.
+            // Items in a row are separated by spaces.
+            Arrays.stream(keysOrder)
+                    .map(dependencyTable::get)
+                    .map(row
+                            -> Arrays.stream(keysOrder)
+                            .map(row == null ? x -> null : row::get)
+                            .map(DependencyValue::toStringS)
+                            .collect(Collectors.joining(" ", "", "\n"))
+                    )
+                    .forEachOrdered((CheckedConsumer<String>) out::append);
+        } catch (final IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        Date finished = new Date();
+        long timeDiff = finished.getTime() - started.getTime();
+        System.getLogger(DependencyMatrixCollector.class.getName())
+                .log(System.Logger.Level.INFO, "Dump finished in {0} ms", timeDiff);
```

Comment:
Which dump? "Dependency matrix dump..." would be clearer.

Reply:
You're right. I didn't think about it.